common/setup.c

```diff
@@ -209,7 +209,7 @@ void __noreturn __text_init kernel_start(uint32_t multiboot_magic, unsigned long
 
     /* Initialize Physical Memory Manager */
     init_regions();
-    init_pmm();
+    init_pmm(mbi_reserved_start(), mbi_reserved_size());
```

Comment:
Maybe instead of passing the values around, create a dedicated (and universal) reserved address range array (similar to the `addr_ranges[]` in `mm/regions.c` and initialize it with the mbi header range during multiboot initialization. Later, other components like PMM can use the reserved ranges array.